modules/fb_instant_articles_display/fb_instant_articles_display.module

```diff
@@ -5,6 +5,132 @@
  * Hook implementations for Facebook Instant Articles Display module.
  */
 
+/**
+ * Load all Field module hooks for Facebook Instant Articles Display module.
+ */
+require_once __DIR__ . '/includes/field.inc';
+
+/**
+ * Implements hook_entity_load().
+ */
+function fb_instant_articles_display_entity_load($entities, $type) {
+  $enabled_entity_types = fb_instant_articles_display_get_article_entity_types();
+  foreach ($entities as $entity) {
+    // Only proceed if the entity is of the configured entity type and bundle.
+    list($id, $vid, $bundle) = entity_extract_ids($type, $entity);
+    if (!in_array($type, array_keys($enabled_entity_types))
+      || !in_array($bundle, array_keys($enabled_entity_types[$type]))) {
+      return FALSE;
+    }
+
+    // We add Entity-specific information to the ArticleWrapper object, and
+    // attach it to the Entity for use in later hooks. Note that we don't store
+    // the InstantArticle object on the node directly, because our Singleton
+    // wrapper + Drupal hook class pattern relies on invoking
+    // hook_fb_instant_articles_article_alter() in ArticleWrapper->getArticle(),
+    // to allow other modules to alter the InstantArticle object whenever it is
+    // requested. If we request that now, it will not have the Entity-specific
+    // information on it this module needs for later altering.
+    $context = array(
+      'entity_type' => $type,
+      'entity' => $entity,
+    );
+    $wrapper = \Drupal\fb_instant_articles\ArticleWrapper::create($context);
```

Comment:
With the hook now running within the call to `ArticleWrapper::create`, if you have a `hook_fb_instant_articles_article_alter` implementation that calls `entity_load`, you trigger an infinite call loop that ends in a Fatal Max Nesting error.  That's b/c this calls the `hook_fb_instant_articles_article_alter` implemenation, which triggers a `entity_load`, which calls the `hook_fb_instant_articles_article_alter`, which triggers an `entity_load`, etc.

I think with this issue and the other issue I noted regarding not being able to override the author, we need to find a place for the alter hook that comes after the Display module is done changing things on the `InstantArticle` object..